### PR TITLE
BUG: Fix environment checking logic for `NUMPY_WARN_IF_NO_MEM_POLICY`

### PR DESCRIPTION
**Pull request by seberg**
The logic was reversed accidentally, since the default is to not warn.

I would be happy to dig a bit deeper and move the `getenv` to import
time (with an small internal helper to set it).  Since the `getenv`
right now will be called for quite a bit of pythran code currently.

---

~I have a bit of a problem with these tests locally (but I am not sure it is specific to these test, had simlar problems elsehwere).  Somehow the wrong `numpy` headers get picked up, even though the current include is passed... (so this is untested locally)~
Tested locally, the test is a bit "unreliably" because it just prints out when the error is raised, but overall that is fine.  My issue with the test locally, is due to debian linking the NumPy includes to: `/usr/include/python3.9/numpy` which probably gets picked up (I guess that is wrong in debian, and I opened a bug-report there).

Edit: Fixup for gh-20200

### CODE REVIEW

**Comment by seberg** [comment]
Hmm, unclear.  The test seems to fail on windows, do the env variables work differently?

**Comment by charris** [comment]
Strange error on pypy that is probably unrelated. Retrying.

**Comment by charris** [comment]
Repeatable error on PyPy.

File: numpy/core/tests/test_mem_policy.py [hunk 1]
```diff
@@ -357,24 +357,36 @@ def test_new_policy(get_module):
     c = np.arange(10)
     assert np.core.multiarray.get_handler_name(c) == orig_policy_name
 
-def test_switch_owner(get_module):
+@pytest.mark.parametrize("policy", ["0", "1", None])
```
Comment (mattip) [inline numpy/core/tests/test_mem_policy.py:360]: I think there is a problem with pytest and os.environ on PyPy. In any case, let's not hold up this PR for that platform.
```suggestion
@pytest.mark.xfail(sys.implementation.name == "pypy",
                   reason=("bad interaction between getenv and "
                           "os.environ inside pytest"))
@pytest.mark.parametrize("policy", ["0", "1", None])
```

Comment (seberg) [reply numpy/core/tests/test_mem_policy.py:363]: May still print out noise, we could set the warning filter to "always", but probably doesn't matter.

**Comment by charris** [comment]
Thanks Sebastian.

**Pull request merged**